### PR TITLE
http: consider cookies over localhost to be secure

### PR DESCRIPTION
**Pull request by bagder**
Updated test31.
Added test 392 to verify secure cookies used for http://localhost

### CODE REVIEW

**Comment by bagder** [comment]
Hm, maybe this is too early. I thought I read that Chrome also did this but according to [this page](https://www.chromestatus.com/feature/6269417340010496), Edge is doing this. The page also says Firefox is only "positive", while Mozilla's [own documentation](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts#when_is_a_context_considered_secure) says they already do this.

File: lib/http.c [hunk 1]
```diff
@@ -2711,14 +2711,16 @@ CURLcode Curl_http_cookies(struct Curl_easy *data,
     int count = 0;
 
     if(data->cookies && data->state.cookie_engine) {
+      const char *host = data->state.aptr.cookiehost ?
+        data->state.aptr.cookiehost : conn->host.name;
+      const bool secure_context =
+        conn->handler->protocol&CURLPROTO_HTTPS ||
+        strcasecompare("localhost", host) ||
+        !strcmp(host, "127.0.0.1") ||
+        !strcmp(host, "[::1]") ? TRUE : FALSE;
```
Comment (danielgustafsson) [inline lib/http.c:2720]: I'm torn, I sort of think this logic belong in `cookie.c` but only if the `secure` parameter instead takes the scheme.. so on the whole maybe this is the cleanest approach.

Comment (bagder) [reply lib/http.c:2720]: Yeah, I'm not totally happy with this big conditional thing like this, and that it is mostly duplicated too, but I couldn't really figure out a better way to do it...

Comment (danielgustafsson) [reply lib/http.c:2720]: I tried but failed as well,  so overall :+1: on this PR.

**Pull request closed without merging**